### ISSUE TITLE
Bug in ROOT volume reflection factory (t_ClientTests_Check_Shape_Eightpoint_Reflect_DetElement FAIL)

### ISSUE THREAD

**Issue by petricm** [AIDASoft/DD4hep#650]
If you execute the test `t_ClientTests_Check_Shape_Eightpoint_Reflect_DetElement` on mac with xerces-c it fails with 

```
        Start  52: t_ClientTests_Check_Shape_Eightpoint_Reflect_DetElement
 52/253 Test  #52: t_ClientTests_Check_Shape_Eightpoint_Reflect_DetElement .......Child aborted***Exception:   1.04 sec
 #### LD_LIBRARY_PATH = :  /cvmfs/sft.cern.ch/lcg/releases/LCG_97python3/ROOT/v6.20.02/x86_64-mac1015-clang110-opt/lib
 ### running test :  'geoDisplay  file:/Users/gitlab-runner/builds/98de57ac/0/CLICdp/DetectorSoftware/DD4hep/examples/examples/ClientTests/compact/Check_Shape_Eightpoint_Reflect_DetElement.xml -load -destroy'
Info in <TGeoManager::TGeoManager>: Geometry default, Detector Geometry created
PersistencyIO    INFO  +++ Set Streamer to dd4hep::OpaqueDataBlock
CompactLoader    INFO  +++ Processing compact file: file:/Users/gitlab-runner/builds/98de57ac/0/CLICdp/DetectorSoftware/DD4hep/examples/examples/ClientTests/compact/Check_Shape_Eightpoint_Reflect_DetElement.xml with flag BUILD_DEFAULT
DD4hep           WARN  ++ STD conditions NOT defined by client. NTP defaults taken.
Info in <TGeoManager::SetTopVolume>: Top volume is world_volume. Master volume is world_volume
Info in <TGeoNavigator::BuildCache>: --- Maximum geometry depth set to 100
Detector         INFO  *********** Created World volume with size:  300  300  300
TestShape        INFO  Created successfull shape of type: EightPointSolid
TestShape        INFO  Correct shape type: EightPointSolid <-> EightPointSolid Instance test: OK
Mesh_Verifier    INFO  +++ Successfully read reference file: /Users/gitlab-runner/builds/98de57ac/0/CLICdp/DetectorSoftware/DD4hep/examples/examples/ClientTests/ref/Ref_Eightpoint_Reflect_DetElement.txt
Mesh_Verifier    INFO  +++ Successfully checked CREATED shapes.
ShapelessSolid   WARN  +++ setDimensions is a compatibility call. Nothing implemented.
Mesh_Verifier    INFO  +++ Successfully checked REDIMENSIONED shapes.
TestShape        INFO  +++ Shape verification SUCCESSFUL. [type=Shape_Trapezoid]
Info in <TGeoManager::CheckGeometry>: Fixing runtime shapes...
Info in <TGeoManager::CheckGeometry>: ...Nothing to fix
Info in <TGeoManager::CloseGeometry>: Counting nodes...
Info in <TGeoManager::Voxelize>: Voxelizing...
Info in <TGeoManager::CloseGeometry>: Building cache...
Info in <TGeoManager::CountLevels>: max level = 3, max placements = 2
Info in <TGeoManager::CloseGeometry>: 6 nodes/ 5 volume UID's in Detector Geometry
Info in <TGeoManager::CloseGeometry>: ----------------modeler ready----------------
Compact          INFO  ++ Converted subdetector:Shape_Trapezoid of type DD4hep_TestShape_Creator 
Detector         INFO  +++ Patching names of anonymous shapes....
The geometry was loaded. Application now exiting.

```

**Comment by MarkusFrankATcernch** [comment]
This fix did not work. Need to re-investigate.

**Comment by petricm** [comment]
I have a bit on new output
```
lcdci@pclcdmac01 examples % geoDisplay  file:/Users/lcdci/DD4hep/examples/examples/ClientTests/compact/Check_Shape_Eightpoint_Reflect_DetElement.xml -load -destroy
PersistencyIO    INFO  +++ Set Streamer to dd4hep::OpaqueDataBlock
Info in <TGeoManager::TGeoManager>: Geometry default, Detector Geometry created
CompactLoader    INFO  +++ Processing compact file: file:/Users/lcdci/DD4hep/examples/examples/ClientTests/compact/Check_Shape_Eightpoint_Reflect_DetElement.xml with flag BUILD_DEFAULT
DD4hep           WARN  ++ STD conditions NOT defined by client. NTP defaults taken.
Detector         INFO  *********** Created World volume with size:  300  300  300
Info in <TGeoManager::SetTopVolume>: Top volume is world_volume. Master volume is world_volume
Info in <TGeoNavigator::BuildCache>: --- Maximum geometry depth set to 100
TestShape        INFO  Created successfull shape of type: EightPointSolid
TestShape        INFO  Correct shape type: EightPointSolid <-> EightPointSolid Instance test: OK
Mesh_Verifier    INFO  +++ Successfully read reference file: /Users/lcdci/DD4hep/examples/examples/ClientTests/ref/Ref_Eightpoint_Reflect_DetElement.txt
Mesh_Verifier    INFO  +++ Successfully checked CREATED shapes.
ShapelessSolid   WARN  +++ setDimensions is a compatibility call. Nothing implemented.
Mesh_Verifier    INFO  +++ Successfully checked REDIMENSIONED shapes.
TestShape        INFO  +++ Shape verification SUCCESSFUL. [type=Shape_Trapezoid]
Compact          INFO  ++ Converted subdetector:Shape_Trapezoid of type DD4hep_TestShape_Creator 
Info in <TGeoManager::CheckGeometry>: Fixing runtime shapes...
Info in <TGeoManager::CheckGeometry>: ...Nothing to fix
Info in <TGeoManager::CloseGeometry>: Counting nodes...
Info in <TGeoManager::Voxelize>: Voxelizing...
Info in <TGeoManager::CloseGeometry>: Building cache...
Info in <TGeoManager::CountLevels>: max level = 3, max placements = 2
Info in <TGeoManager::CloseGeometry>: 6 nodes/ 5 volume UID's in Detector Geometry
Info in <TGeoManager::CloseGeometry>: ----------------modeler ready----------------
Detector         INFO  +++ Patching names of anonymous shapes....
The geometry was loaded. Application now exiting.
geoDisplay(19106,0x107035dc0) malloc: *** error for object 0x7f8fb953e7d0: pointer being freed was not allocated
geoDisplay(19106,0x107035dc0) malloc: *** set a breakpoint in malloc_error_break to debug
zsh: abort      geoDisplay  -load -destroy
```
this malloc problem only appears if you execute the test manually and not via ctest

**Comment by petricm** [comment]
The core cause of failing test is a [bug in TGeo](https://sft.its.cern.ch/jira/browse/ROOT-10914), needs to be addressed by the ROOT team.

**Comment by petricm** [comment]
The issues has been resolve in https://github.com/root-project/root/commit/437486008912b2da317097eaf7ea8f2ea1a822c4 . As this affects only the destruction of reflected shapes I would suggest to @mrodozov @ianna @cvuosalo if you have the possibility to cherry-pick this fix into your ROOT build until a patch release of ROOT includes this. We did not observe that this bug manifested itself on CentOS7, but it does on macOS and I don't know on what other platforms you are running ... so depending on the platform you might or might not be affected.

**Comment by mrodozov** [comment]
we are running on slc6 cc7 and cc8, but we have one piece that uses ROOT running on Mac (Fireworks)
The commit is in master now so it will be integrated in our ROOT master IB. Besides ROOT master however we use also 6-20 and 6-22. I'll see if we need this backported there

**Comment by mrodozov** [comment]
the commit is included in both 6-20 and 6-22

**Comment by MarkusFrankATcernch** [comment]
@mrodozov For completeness: The bug (double delete, in case the reflected shape was an assembly) was present on all platforms. However, it did manifest itself with an exception only on MAC....

**Comment by petricm** [comment]
This has been resolved in ROOT versions 6.20/08 or higher patch, or 6.22/02 or higher (6.22/00 does not have the fix)
